File: check_balance.js

const fs = require('fs');
const filepath = 'd:/GATE MEET/SportBuzz/SportBuzz/src/components/admin/TournamentManager.tsx';
const content = fs.readFileSync(filepath, 'utf8');
const lines = content.split('\n');

const startLine = 1217;
const endLine = 2260;
const block = lines.slice(startLine - 1, endLine).join('\n');

let parens = 0;
let braces = 0;
let angleBrackets = 0;

for (let i = 0; i < block.length; i++) {
    const char = block[i];
    if (char === '(') parens++;
    else if (char === ')') parens--;
    else if (char === '{') braces++;
    else if (char === '}') braces--;
    else if (char === '<' && block[i+1] !== ' ' && block[i+1] !== '=') angleBrackets++;
    else if (char === '>' && block[i-1] !== ' ' && block[i-1] !== '-') angleBrackets--;
}

console.log('Analysis for lines ' + startLine + '-' + endLine + ':');
console.log('Parentheses balance:', parens);
console.log('Braces balance:', braces);
console.log('Angle brackets balance (rough):', angleBrackets);

File: check_balance.py

import os

filepath = r'd:\GATE MEET\SportBuzz\SportBuzz\src\components\admin\TournamentManager.tsx'
with open(filepath, 'r', encoding='utf-8') as f:
    lines = f.readlines()

start = 1217
end = 2260
content = "".join(lines[start-1:end])

paren_count = 0
brace_count = 0
for char in content:
    if char == '(': paren_count += 1
    elif char == ')': paren_count -= 1
    elif char == '{': brace_count += 1
    elif char == '}': brace_count -= 1

print(f"Paren count (1217-2260): {paren_count}")
print(f"Brace count (1217-2260): {brace_count}")
